src/studio/src/designer/frontend/shared/components/FileSelector.test.tsx

```diff
@@ -30,6 +30,34 @@ describe('FileSelector', () => {
       'hello.png',
     );
   });
+
+  it('Should show text on the button by default', async () => {
+    render();
+    expect(screen.getByText('Upload button text')).toBeInTheDocument();
+  });
+
+  it('Should not show text on the button when it is part of the top toolbar', async () => {
+    render({ isInTopToolbar: true });
+    expect(screen.queryByText('Upload button text')).not.toBeInTheDocument();
+  });
+
+  it('Should call file input onClick handler when the default upload button is clicked', async () => {
+    render();
+    const button = screen.getByText('Upload button text');
+    const fileInput = screen.getByTestId('FileSelector-input');
+    fileInput.onclick = jest.fn();
```

Comment:
The reason why I have chosen to do it this way is that the uploading consists of two steps; first, the user clicks the button to open the explorer window, and then he or she picks a file and uploads it. This test checks the first step, while `should call submitHandler when a file is selected` checks the first one. With a normal file selector, I guess that this test wouldn't be necessary, but here, there is a button that calls the `onClick` handler on another, invisible button. That is all this test is supposed to check, as the rest is covered by the other one. I agree that it would be better to merge these events into one single test, but I couldn't find a good way to do so.

Reply:
Ah, ok I think I've misunderstood what was going on here the first time I read it. But not sure if this is really suited as a unit test. It shouldnt know about any internals, and the test may break while doing changes to the internals.

Perhaps this is better suited to cover in a Cypress test?

Reply:
I can't find a better way to do this with Cypress. I have tried some methods using event listeners, but I couldn't make it work, and it didn't seem to be possible without in any way targeting the hidden file input specifically in the code. Also, Cypress recommends targeting the hidden input directly in their own documentation; https://docs.cypress.io/api/commands/selectfile#On-a-hidden-input. Their example basically does the same thing as `should call submitHandler when a file is selected`.

I'll merge this now, but let me know if you have any other suggestions.